### PR TITLE
NEXUS-34974 Fixing index not being updated

### PR DESCRIPTION
**Pull request by userSeprid**
#### Description of Change
Fetch the nexus repository tgz files before the `helm repo index` call

#### Things to Do Before Submitting

* Have you signed the [Sonatype CLA](https://sonatypecla.herokuapp.com/sign-cla)?
* Have you added and run automated tests for your change? 
  (See the [README](https://github.com/sonatype/helm3-charts/blob/main/README.md))
* Have you run `helm lint` on your change?

### CODE REVIEW

File: Jenkinsfile-Release
Comment (bobotimi) [inline Jenkinsfile-Release:65]: I'd rename build.sh to build-nexus.sh then do as follows:
`
 if ('nexus-repository-manager'.equals(params.chart)) {
      runSafely "curl -L 'https://github.com/sonatype/nxrm3-helm-repository/blob/main/docs/${nexusRepoManagerArtifactName}?raw=true' -o 'docs/${nexusRepoManagerArtifactName}'"
      runSafely "curl -L 'https://github.com/sonatype/nxrm3-helm-repository/blob/main/docs/${awsResiliencyArtifactName}?raw=true' -o 'docs/${awsResiliencyArtifactName}'"
      runSafely './build-nexus.sh'
    }
`

Comment (bobotimi) [reply Jenkinsfile-Release:65]: Essentially for nexus, you need to run the curl before the helm repo index command contained in the build.sh (to be renamed build-nexus.sh) file

File: build-iq.sh
Comment (jflinchbaugh) [inline build-iq.sh:30]: if you remove this from here, it seems that it would _not_ index when IQ releases. I think it should stay here.

File: Jenkinsfile-Release
Comment (jflinchbaugh) [inline Jenkinsfile-Release:70]: this block also exists below (line ~75 or so). can it be removed from there?

Comment (bobotimi) [reply Jenkinsfile-Release:70]: Ah yes, that needs to come out

File: build-iq.sh
Comment (bobotimi) [reply build-iq.sh:30]: Yes that should be there -  I didn't know that was taken out.

File: Jenkinsfile-Release
Comment (bobotimi) [reply Jenkinsfile-Release:70]: https://github.com/sonatype/helm3-charts/pull/245/commits/70dde6c411cfee6d1e51f5e5e455ccdbe2ac0052

File: build-iq.sh
Comment (bobotimi) [reply build-iq.sh]: https://github.com/sonatype/helm3-charts/pull/245/commits/5c4fb48befcfc8fd94a7fcd8f3331064a3d76380

**Review by jflinchbaugh — APPROVED** [review]
+1

**Comment by jflinchbaugh** [comment]
If i remember correctly, I probably need to merge this for you when you get a +1 from your team?

**Comment by jflinchbaugh** [comment]
Oops, the build is broken: https://jenkins.ci.sonatype.dev/job/integrations/job/cloud/job/Helm3%20Charts/job/Feature%20Snapshot%20Builds/job/NEXUS-34974-fixing-index-not-being-updated/4/console

The Jenkinsfile for snapshot builds has a reference to the old "build.sh", so that needs to be updated.

**Comment by bobotimi** [comment]
> Oops, the build is broken: https://jenkins.ci.sonatype.dev/job/integrations/job/cloud/job/Helm3%20Charts/job/Feature%20Snapshot%20Builds/job/NEXUS-34974-fixing-index-not-being-updated/4/console
> 
> The Jenkinsfile for snapshot builds has a reference to the old "build.sh", so that needs to be updated.

Done

**Comment by userSeprid** [comment]
+1

**Comment by bobotimi** [comment]
> Oops, the build is broken: https://jenkins.ci.sonatype.dev/job/integrations/job/cloud/job/Helm3%20Charts/job/Feature%20Snapshot%20Builds/job/NEXUS-34974-fixing-index-not-being-updated/4/console
> 
> The Jenkinsfile for snapshot builds has a reference to the old "build.sh", so that needs to be updated.

Fixed

**Pull request merged**
